devbox.go

```diff
@@ -91,7 +91,8 @@ func (d *Devbox) Build(opts ...docker.BuildOptions) error {
 // environment.
 func (d *Devbox) Plan() *planner.BuildPlan {
 	basePlan := &planner.BuildPlan{
-		Packages: d.cfg.Packages,
+		Packages:  d.cfg.Packages,
+		ShellHook: d.cfg.ShellHook,
```

Comment:
I'm assuming `ShellHook` is only applicable on `devbox shell` command, and not `devbox build`? If so, we don't need to add it to the planner

Reply:
@LucilleH `Plan` is currently called by both `devbox shell` and `devbox build`

Reply:
Oh so in this case the shell hook will be run in both scenario. Ok that's a bit different from https://github.com/jetpack-io/devbox/issues/42 since that ask is for shell only

Reply:
@loreto sorry to clarify, the shell hook in this case won’t actually be called in the Dockerfile?

Reply:
It won't run because there's nothing that executes it. This behaves as you want. I'm just saying the same struct that includes the hook is passed to both shell and build.